### PR TITLE
Function to validate if string is valid 'label id'

### PR DESCRIPTION
**Pull request by AlexHimself**
When creating a new "label id", there is a specific string format used. This function simply allows you to pass a string and return if it is a valid "label id" format.

Developer will figure out their own unique uses for this. I used it as a helper function to generate correctly formatted label texts to save time creating them.

See image for further clarification. I decompiled an assembly used by the D365 Editor Extensions to determine the Regex.

image

### CODE REVIEW

File: d365fo.tools/functions/get-d365labelidvalid.ps1 [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        Checks if a string is a valid 'Label Id' format
+ 
+    .DESCRIPTION
+        This function will validate if a string is a valid 'Label Id' format.
+
+    .PARAMETER LabelId
+        LabelId string to validate
+ 
+    .EXAMPLE
+        Get-D365LabelIdIsValid -LabelId "ABC123" #True
```
Comment (Splaxi) [inline d365fo.tools/functions/get-d365labelidvalid.ps1]: We always prefix an example like this:
PS C:\> Get-D365LabelIdIsValid...

This is to make it feel more like the real thing when beginner are trying to learn how to use the module.

File: d365fo.tools/functions/get-d365labelidvalid.ps1 [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        Get-D365LabelIdIsValid -LabelId "@ABC123" #True
+        Get-D365LabelIdIsValid -LabelId "@ABC123_1" #False
+        Get-D365LabelIdIsValid -LabelId "ABC.123" #False
```
Comment (Splaxi) [inline d365fo.tools/functions/get-d365labelidvalid.ps1]: We would like to have each and every single example to be in a separate .EXAMPLE section.

This is both from a user perspective and from a testing perspective.

For each example section we would like that you describe what the example does and what to expect from it, written in a meaningful and descriptive way.

You can get some inspiration from most functions, but this one shows in a good way what we are trying to do:
https://github.com/d365collaborative/d365fo.tools/blob/development/docs/Get-D365Module.md

**Comment by Splaxi** [comment]
Is the name easy enough for outsiders to understand what it does?

Could Test-D365LabelIdValidity or Test-D365LabelIdNamingConvention be better alternatives? Or something else? Please read the list of approved verbs, to see if there are other verbs that might make more sense:

https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-6

**Comment by AlexHimself** [comment]
Let's see if I did this PR correctly

**Comment by AlexHimself** [comment]
Trying again...UTF8 encoding was specified in VSCode, no clue why it was encoded as ASCII though?

**Comment by AlexHimself** [comment]
What am I doing wrong?

**Comment by AlexHimself** [comment]
I think I might have screwed up a little and added a new function to this branch instead of creating a new branch & PR. I need to do some reading on how git works...is this fixable or should I just delete everything and try again? It's just 2 files I'm trying to add so that might be easier.

**Comment by Splaxi** [comment]
I did up take your PR, but I broke things during so... Will update with the status of the merge later.

**Comment by Splaxi** [comment]
Please read through the code from the other PR:

https://github.com/d365collaborative/d365fo.tools/pull/356

You should also be able to download the code and run some test runs against it, if you want.

**Comment by Splaxi** [comment]
Things have been released with the 0.5.64 release.

Give it a test swing and see if things are working like expected.

**Pull request closed without merging**